packages/grafana-runtime/src/services/SidecarService_EXPERIMENTAL.ts

```diff
@@ -20,14 +30,35 @@ import { HistoryWrapper, LocationService, locationService } from './LocationServ
 export class SidecarService_EXPERIMENTAL {
   private _initialContext: BehaviorSubject<unknown | undefined>;
   private memoryLocationService: LocationService;
-  private history: LocationStorageHistory;
+  private mainLocationService: LocationService;
+  private follow = false;
+  private mainLocationWhenOpened: string | undefined;
+  private mainOnAllowedRoute = false;
```

Comment:
Not sure about the overall code structure, will probably try to simplify it a bit if the behaviors is good in general.